Mathlib/Data/Set/Finite/Lattice.lean

```diff
@@ -152,6 +152,42 @@ theorem Finite.iUnion {ι : Type*} {s : ι → Set α} {t : Set ι} (ht : t.Fini
   · rw [he i hi, mem_empty_iff_false] at hx
     contradiction
 
+/-- An indexed union of pairwise disjoint sets is finite iff all sets are finite, and all but
+finitely many are empty. -/
+lemma PairwiseDisjoint.finite_biUnion_iff {f : β → Set α} {s : Set β}
+    (hs : s.PairwiseDisjoint f) :
+    Set.Finite (⋃ i ∈ s, f i)
```

Comment:
I'd liked to have seen this be of the more general form
```suggestion
    Set.Finite (⋃ (i) (h : i ∈ s, f i h)
```
but I guess the definition of `PairwiseDisjoint` doesn't permit this; so no action needed.